tests/installation/grub_utils.pm

```diff
@@ -0,0 +1,82 @@
+package grub_utils;
+
+use base Exporter;
+use Exporter;
+
+use strict;
+use warnings;
+use opensusebasetest qw(handle_uefi_boot_disk_workaround);
+use testapi;
+use utils;
+use version_utils qw(is_sle is_livecd);
+use bootloader_setup qw(stop_grub_timeout boot_into_snapshot);
+
+our @EXPORT = qw(
+  grub_test
+);
+
+=head2 handle_installer_medium_bootup
+
+Due to pre-installation setup, qemu boot order is always booting from CD-ROM.
+=cut
+sub handle_installer_medium_bootup {
+    return unless (check_var("BOOTFROM", "d") || (get_var('UEFI') && get_var('USBBOOT')));
+    assert_screen 'inst-bootmenu', 180;
+
+    if (check_var("BOOTFROM", "d") && check_var("AUTOUPGRADE") && check_var("PATCH")) {
+        assert_screen 'grub2';
+    }
+
+    # Layout of live is different from installation media
+    my $key = is_livecd() ? 'down' : 'up';
+    send_key_until_needlematch 'inst-bootmenu-boot-harddisk', $key;
+    send_key 'ret';
+
+    # use firmware boot manager of aarch64 to boot upgraded system
+    handle_uefi_boot_disk_workaround() if (check_var('ARCH', 'aarch64'));
+}
+
+sub bug_workaround_bsc1005313 {
+    record_soft_failure "Running with plymouth:debug to catch bsc#1005313" if get_var('PLYMOUTH_DEBUG');
+    send_key 'e';
+    # Move to end of kernel boot parameters line
+    send_key_until_needlematch "linux-line-selected", "down";
+    send_key "end";
+
+    assert_screen "linux-line-matched";
+    if (get_var('PLYMOUTH_DEBUG')) {
+        # remove "splash=silent quiet showopts"
+        for (1 .. 28) { send_key "backspace" }
+        type_string 'plymouth:debug';
+    }
+    type_string " " . get_var('GRUB_KERNEL_OPTION_APPEND') if get_var('GRUB_KERNEL_OPTION_APPEND');
+
+    save_screenshot;
+    send_key 'ctrl-x';
+}
+
+sub grub_test {
+    my $timeout = get_var('GRUB_TIMEOUT', 90);
```

Comment:
mind adding a very brief doc? so that it's shown at: https://os-autoinst.github.io/os-autoinst-distri-opensuse/ once the lib is moved to lib/

Reply:
Added the doc and moved `grub_test` to lib. VR passed: https://openqa.suse.de/tests/5834135